### PR TITLE
Merge last netmap github code to FreeBSD tree: FreeBSD 10.3-STABLE-r305972M and FreeBSD-11-RC3

### PR DESCRIPTION
**Pull request by alexhebra**
Hi,

Here is the patchs for both FreeBSD 10.3-STABLE-r305972M and FreeBSD-11-RC3. 
Thanks.

[FreeBSD-10.3-STABLE-r305972M.diff.zip](https://github.com/luigirizzo/netmap/files/490596/FreeBSD-10.3-STABLE-r305972M.diff.zip)
[FreeBSD-11.0-RC3.diff.zip](https://github.com/luigirizzo/netmap/files/490620/FreeBSD-11.0-RC3.diff.zip)

### CODE REVIEW

**Comment by vmaffione** [comment]
Hi,
  The pull request does not contain your code, but commits that are already in the repository, so we cannot use it.

Maybe there has been a bit of misunderstanding (sorry for that), because I see that the patches you provided are (obviouslly huge patches) based on what it is currently on the FreeBSD tree (10.3 or 11). Since we can easily generate these kind of patches by ourselves, they are not particularly useful to us. 

What it would make sense to accept here is just patches against the netmap version on the current 
master branch of this github repository. The purpose of these patches (if any) would be to fix compilation problems and bugs that show up in FreeBSD-CURRENT, so that we can later update the FreeBSD-CURRENT code in the SVN repository.

In other words, once we make sure that the master branch compiles fine on FreeBSD-CURRENT, what we really need is volunteers testing this code on FreeBSD-CURRENT, in the various configurations allowed by netmap:

* netmap on physical NICs with native netmap support (i.e. ix, igb, etc)
* netmap on physical NICs or software devices (e.g TAPs, bridges interfaces, etc), to test the generic/emulated netmap adapter
* netmap pipes
* netmap monitors
* netmap passthrough in bhyve (if_ptnet).

Your help is welcome here!

**Comment by alexhebra** [comment]
Ok, sorry for that. FreeBSD netmap code is too old (more than one year ago). What i was trying to help was generating a patch syncing last github code and FreeBSD source tree. So if you are a FreeBSD committer you would use this patch to update there. As Luigi sent to maillist freebsd-net@ asking for help, i thought it would be useful.  Would you like i send a patch through freebsd website (send-pr)?

About the tests needed, it can be done only on FreeBSD-Current? or can i test it on FreeBSD-11 too?

Thanks.

**Comment by vmaffione** [comment]
We know, the code in FreeBSD is very old. I'm not a commiter, but Luigi is.
Thanks for the patch, but what Luigi meant was to help to make sure netmap is working on FreeBSD Current, so that it can be merged there.

This task requires two steps:
1) Check that it compiles on FreeBSD current. This step should be completed, as you confirmed the compilation is working. As a consequence, we don't need any patch. It is easy for us to generate the patch, and we will do that when step 2 below is completed.
2) Check that the functionalities I listed in my previous mail are working. Here is where you can help if you wish.

Since we are going to merge the code in FreeBSD current it is not so much important to test the code in FreeBSD-11. Testing it in FreeBSD current is what it really counts.

**Comment by alexhebra** [comment]
Ok i'll help with step 2. I'll update my system to FreeBSD-CURRENT and let you know the results, but i gonna need some days. :)

**Comment by vmaffione** [comment]
Sure, take your time. Thanks.
